workflows/iuf/operations/vcs-upload-content.yaml

```diff
@@ -0,0 +1,172 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: vcs-upload-content-main-steps
+spec:
+  entrypoint: main
+  arguments:
+    parameters:
+      - name: name
+      - name: version
+      - name: content
+  templates:
+### Main Steps ###
+  - name: main
+    steps:
+    - - name: get-vcs-secrets
+        template: get-vcs-secrets-template
+    - - name: gitea-upload-content
+        template: gitea-upload-content
+        arguments:
+          parameters:
+          - name: vcs_user_credentials_secret_name
+            value: "{{steps.get-vcs-secrets.outputs.parameters.secret_name}}"
+    - - name: cleanup
+        template: cleanup-template
+        arguments:
+          parameters:
+          - name: vcs_user_credentials_secret_name
+            value: "{{steps.get-vcs-secrets.outputs.parameters.secret_name}}"
+### Templates ###
+## get-vcs-secrets-template ##
+  - name: get-vcs-secrets-template
+    nodeSelector:   
+      kubernetes.io/hostname: ncn-m001
+    tolerations:
+    - key: node-role.kubernetes.io/master
+      effect: NoSchedule
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+    outputs:
+      parameters:
+        - name: secret_name
+          valueFrom:
+            path: /tmp/secret_name
+    retryStrategy:
+        limit: "2"
+        retryPolicy: "Always"
+        backoff:
+          duration: "10s" # Must be a string. Default unit is seconds. Could also be a Duration, e.g.: "2m", "6h", "1d"
+          factor: "2"
+          maxDuration: "1m"
+    script:
+      image: artifactory.algol60.net/csm-docker/stable/docker.io/portainer/kubectl-shell:latest-v1.21.1-amd64
+      command: [bash]
+      source: |
+        function sync_item() {
+          item_name="$1"
+          source_ns="$2"
+          destination_name="$3-$RANDOM"
+          destination_ns="$4"
+          if kubectl get $item_name -n $source_ns &> /dev/null; then
+            echo "Syncing $item_name from $source_ns to $destination_ns as $destination_name"
+            kubectl get $item_name -n $source_ns -o json | \
+              jq 'del(.metadata.namespace)' | \
+              jq 'del(.metadata.creationTimestamp)' | \
+              jq 'del(.metadata.resourceVersion)' | \
+              jq 'del(.metadata.selfLink)' | \
+              jq 'del(.metadata.uid)' | \
+              jq 'del(.metadata.ownerReferences)' | \
+              jq 'del(.metadata.name)' | \
+              jq '.metadata |= . + {"name":"'$destination_name'"}' | \
+              kubectl apply -n $destination_ns -f -
+              return $?
+          else
+            echo "Didn't find $item_name in the $source_ns namespace"
+            return 1
+          fi
+        }
+        sync_item secret/vcs-user-credentials services vcs-user-credentials-argo argo
```

Comment:
So, we're just copying the secret from the services name space over to the argo namespace. Is it copied over as a secret? I wonder how the argo namespace permissions different from the services namespace permissions. We should run this by @jeremy-duckworth .